coiote-aws-iot-cloud-formation/lwm2mOperation/lambda_function.py

```diff
@@ -1,17 +1,38 @@
 import json
-import requests
-import boto3
+import os
+import tempfile
 from dataclasses import dataclass
 
+import boto3
+import requests
+
+REST_URI = os.environ['coioteDMrestUri'] + '/api/coiotedm/v3'
+
+
+def getCertificateFiles():
+    userAuthCert = getCoioteSecrets()
+    certificatePem = tempfile.NamedTemporaryFile(delete=False)
+    privateKey = tempfile.NamedTemporaryFile(delete=False)
+    certificatePemFile = certificatePem.name
+    privateKeyFile = privateKey.name
+    try:
+        certificatePem.write(str.encode(userAuthCert.certificatePem))
+        privateKey.write(str.encode(userAuthCert.privateKey))
+        certificatePem.close()
+        privateKey.close()
+        yield certificatePemFile, privateKeyFile
+    finally:
+        os.unlink(certificatePemFile)
+        os.unlink(privateKeyFile)
```

Comment:
Całkiem sprytne yieldowanko. Zastanawiam się, czy coś takiego 
```suggestion
def getCertificateFiles():
    with tempfile.NamedTemporaryFile() as cert, tempfile.NamedTemporaryFile() as pk:
        userAuthCert = getCoioteSecrets()
        cert.write(str.encode(userAuthCert.certificatePem))
        pk.write(str.encode(userAuthCert.privateKey))
        yield cert.name, pk.name
```
nie zrobiłoby sztuczki, z mojego niezależnego testu wynika, że nawet jak wyrzuci jakiegoś exceptiona, to pliki są kasowane, a nie trzeba robić tego wszystkiego

Reply:
Jeszcze dodam, mimo że stosujemy w tym pliku luźną konwencję pajtonową (xd), to pisałbym te zmienne snake casem